planner/core/logical_plan_builder.go

```diff
@@ -2426,7 +2426,7 @@ func (b *PlanBuilder) buildProjUponView(ctx context.Context, dbName model.CIStr,
 		}
 		projSchema.Append(&expression.Column{
 			UniqueID:    b.ctx.GetSessionVars().AllocPlanColumnID(),
-			TblName:     col.TblName,
+			TblName:     tableInfo.Name,
```

Comment:
Do we need to change `DBName`?

Reply:
@XuHuaiyu PTAL. BTW, we can merge this PR firstly. Changing the `DBName` in another PR if needed.

Reply:
Nice catch.
https://github.com/pingcap/tidb/issues/15611